### PR TITLE
[Native][AmplifyStorage] doc updates

### PR DESCRIPTION
**Pull request by DongQuanRui**
*Description of changes:*
- Update iOS code snippets to use Amplify.Auth
- Update iOS code snippets to use new Async-Event-handler.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/storage/fragments/ios/download.md [hunk 1]
```diff
@@ -5,16 +5,15 @@ There are three ways of getting data that was previously uploaded:
...
-        print("Completed: \(data)")
-    case let .failed(storageError):
-        print("Failed: \(storageError.errorDescription). \(storageError.recoverySuggestion)")
-    case let .inProcess(progress):
-        print("Progress: \(progress)")
-    default:
-        break
+_ = Amplify.Storage.downloadData(key: "myKey", 
+    progressListener: { progress in
+        print(progress)
```
Comment (lawmicha) [inline docs/lib/storage/fragments/ios/download.md:10]: `print("Progress: \(progress)")`

File: docs/lib/storage/fragments/ios/getting-started.md [hunk 2]
```diff
@@ -98,7 +98,6 @@ Add the following imports to the top of your `AppDelegate.swift` file:
 
 ```swift
 import Amplify
-import AWSMobileClient
 import AmplifyPlugins
 ```
```
Comment (lawmicha) [inline docs/lib/storage/fragments/ios/getting-started.md:103]: looks like this entire file can be removed now that it is using the `getting-started/common.md`.

File: docs/lib/storage/fragments/ios/getting-started/40_upload.md [hunk 3]
```diff
@@ -2,17 +2,16 @@
...
-            print("Completed: \(data)")
-        case .failed(let storageError):
-            print("Failed: \(storageError.errorDescription). \(storageError.recoverySuggestion)")
-        case .inProcess(let progress):
-            print("Progress: \(progress)")
-        default:
-            break
+    _ = Amplify.Storage.uploadData(key: "myKey", data: data, 
+        progressListener: { progress in
+            print(progress)
```
Comment (lawmicha) [inline docs/lib/storage/fragments/ios/getting-started/40_upload.md:7]: `print("Progress: \(progress)")`

File: docs/lib/storage/fragments/ios/upload.md [hunk 4]
```diff
@@ -3,18 +3,17 @@ To upload to S3 from a data object, specify the `key` and the `data` object to b
...
-        print("Failed: \(storageError.errorDescription). \(storageError.recoverySuggestion)")
-    case let .inProcess(progress):
-        print("Progress: \(progress)")
-    default:
-        break
-    }
-}
+Amplify.Storage.uploadData(key: "myKey", data: data, 
+    progressListener: { progress in
+        print(progress)
```
Comment (lawmicha) [inline docs/lib/storage/fragments/ios/upload.md:8]: `print("Progress: \(progress)")`

File: docs/lib/storage/fragments/ios/upload.md [hunk 5]
```diff
@@ -32,14 +31,10 @@ do {
...
 _ = Amplify.Storage.uploadFile(key: fileNameKey, local: filename) { event in
     switch event {
-    case let .completed(data):
+    case let .success(data):
         print("Completed: \(data)")
-    case let .failed(storageError):
+    case let .failure(storageError):
         print("Failed: \(storageError.errorDescription). \(storageError.recoverySuggestion)")
-    case let .inProcess(progress):
-        print("Progress: \(progress)")
```
Comment (lawmicha) [inline docs/lib/storage/fragments/ios/upload.md:40]: can you add `progressListener` to this one too?

File: docs/lib/storage/fragments/ios/getting-started.md (hunk 2 above)
Comment (DongQuanRui) [reply docs/lib/storage/fragments/ios/getting-started.md:103]: Yeah, I have removed this getting-started from Analytics, Haven't removed here. Going to remove it

File: docs/lib/storage/fragments/ios/upload.md (hunk 5 above)
Comment (DongQuanRui) [reply docs/lib/storage/fragments/ios/upload.md:40]: Added

File: docs/lib/storage/fragments/ios/upload.md (hunk 4 above)
Comment (DongQuanRui) [reply docs/lib/storage/fragments/ios/upload.md:8]: changed

File: docs/lib/storage/fragments/ios/getting-started/40_upload.md (hunk 3 above)
Comment (DongQuanRui) [reply docs/lib/storage/fragments/ios/getting-started/40_upload.md:7]: changed

File: docs/lib/storage/fragments/ios/download.md (hunk 1 above)
Comment (DongQuanRui) [reply docs/lib/storage/fragments/ios/download.md:10]: changed

**Comment by lawmicha** [comment]
the conflict
```
<<<<<<< next-storage
    _ = Amplify.Storage.uploadData(key: "myKey", data: data, 
        progressListener: { progress in
=======
    
    Amplify.Storage.uploadData(key: "ExampleKey", data: data) { (event) in
        switch event {
        case .completed(let data):
            print("Completed: \(data)")
        case .failed(let storageError):
            print("Failed: \(storageError.errorDescription). \(storageError.recoverySuggestion)")
        case .inProcess(let progress):
>>>>>>> next
``` 

you will want to take the first section that is coming from `next-storage` branch

**Pull request merged**